omnigibson/objects/dataset_object.py

```diff
@@ -548,6 +564,9 @@ def get_base_aligned_bbox(self, link_name=None, visual=False, xy_aligned=False,
                 # Add the points to our collection of points.
                 points.extend(trimesh.transformations.transform_points(vertices_in_base_frame, bbox_center_in_base_frame))
             elif fallback_to_aabb:
+                # If we're visual and the mesh is not visible, there is no fallback so continue
+                if bbox_type == "visual" and not np.all(tuple(mesh.visible for mesh in meshes.values())):
+                    continue
```

Comment:
Remove 
```
                if bbox_type == "visual" and "visual" not in self.native_link_bboxes[link_name]:
                    log.debug(
                        "Falling back to collision bbox for object %s link %s since no visual bbox exists.",
                        self.name,
                        link_name,
                    )
                    bbox_type = "collision"
```

Reply:
Check if this is still needed:
```
                # TODO: Because of encoding, may need to be UsdTokens.none, not None
                if self.native_link_bboxes[link_name][bbox_type] is None:
                    continue
```

Reply:
removed